src/client-side-encryption/auto_encrypter.ts

```diff
@@ -237,6 +245,16 @@ export class AutoEncrypter {
     this._proxyOptions = options.proxyOptions || {};
     this._tlsOptions = options.tlsOptions || {};
     this._kmsProviders = options.kmsProviders || {};
+    this._credentialProviders = options.credentialProviders;
+
+    if (
+      options.credentialProviders?.aws &&
+      !isEmptyCredentials('aws', options.kmsProviders || {})
+    ) {
+      throw new MongoCryptInvalidArgumentError(
+        'Cannot provide both a custom credential provider and credentials. Please specify one or the other.'
```

Comment:
```suggestion
        'Can only provide a custom AWS credential provider when the state machine is configured for automatic AWS credential fetching.'
```
`isEmptyCredentials` returns false if there is no configuration for the provider as well, so credentials like this would trigger this error to be thrown:

```typescript
{
  kmsProviders: { azure: {} },
  credentialProviders: { aws: () => .... }
}
```

Reply:
Updated.